### DISCUSSION TITLE
Customising Keycloak logging / Question around overall logging architecture of Keycloak / Use another logging framework apart JBoss Logging

### DISCUSSION THREAD

**Discussion by Laurenzi** [keycloak/keycloak#22706] [Q&A]
Hello,
I see that Keycloak documentation mentions that Keycloak uses JBoss Logging Framework for logging.

What I'd like to understand is the following:

1. What is the overall Logging architecture in Keycloak
2. Is it possible to use other logging frameworks such as Log4J2 or Logback and if so, how to do this?
3. Is it possible to somehow customise the stack trace that is printed out by Keycloak? See also StackOverflow question here: https://stackoverflow.com/questions/76951913/how-to-customise-how-keycloak-logs-stack-traces

Thank you!

**Comment by hppzone8** [comment]
I have also same question, is it possible to integrate logback ?

**Comment by shawkins** [comment]
The short answer is this is not supported. The longer answer is that you may try to customize keycloak with additional quarkus extensions. See https://quarkus.io/extensions/io.quarkiverse.logging.logback/quarkus-logging-logback/ and https://github.com/mabartos/keycloak-quarkus-extensions
